### PR TITLE
Integrating into Kubescape network Scanner using the Package structure

### LINKED ISSUE #2: Moving the code under this repository and align it with the package structre

### PR DESCRIPTION
**Pull request by 0xquark**
Fixes #2 

I've structured the code using the go styling guide. We have a `networkscanner` package under which we have subpackages `portdiscovery` and `servicediscovery`. `servicediscovery` package is divided into more subpackage like `applicationlayerdiscovery`. This make the structure easy to use. 
```
.
├── LICENSE
├── README.md
├── cmd
│   └── main.go
├── go.mod
└── internal
    └── pkg
        └── networkscanner
            ├── interface.go
            ├── portdiscovery
            │   └── portdiscovery.go
            └── servicediscovery
                ├── applicationlayerdiscovery
                │   ├── al_list.go
                │   └── al_mysql.go
                ├── presentationlayerdiscovery
                │   ├── pl_httpdiscovery.go
                │   └── pl_list.go
                ├── sessionlayerdiscovery
                │   ├── sl_list.go
                │   ├── sl_tcp.go
                │   └── sl_tls.go
                └── types.go
```
Most of the code is integrated now and CLI will be integrated in a different PR.

### CODE REVIEW

**Comment by 0xquark** [comment]
@slashben Could you check if this layout is appropriate and aligns with kubescape?

**Comment by slashben** [comment]
@Bezbran can this project layout work?

**Comment by Bezbran** [comment]
Hi @0xquark, can you please ellaborate about the `internal/pkg/network-scanner` path? Why not put the content of this path on the root of the repository?
You can take a look on [kubescape repo](https://github.com/kubescape/kubescape), it seems well organized.

**Comment by 0xquark** [comment]
> Hi @0xquark, can you please ellaborate about the `internal/pkg/network-scanner` path? Why not put the content of this path on the root of the repository? You can take a look on [kubescape repo](https://github.com/kubescape/kubescape), it seems well organized.

Hi @Bezbran ! The `internal/pkg/network-scanner` was chosen by ben ( CC : @slashben  ) for which i believe the right reason was to having a separate `internal/pkg` directory is to separate internal packages that are not meant to be exposed to users or other packages outside of the project.

File: internal/pkg/networkscanner/portdiscovery/portdiscovery.go [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+}
+
+type ScanTarget struct {
+	Hostname string
+	IPStart  net.IP
+}
+
+// Struct defining the result of the network scanner
+
+func parseArgs() (*ScanConfig, error) {
```
Comment (slashben) [inline internal/pkg/networkscanner/portdiscovery/portdiscovery.go:31]: Looks like part of the CLI (should be moved there)

**Review by slashben — COMMENTED** [review]
.DS_Store files should be removed (added to .gitignore as well)

**Comment by 0xquark** [comment]
Hey @slashben ! Can you review the changes in an hour? I am making some final fixes then this PR would be ready to review

**Comment by slashben** [comment]
@0xquark working on it!

File: internal/pkg/networkscanner/portdiscovery/portdiscovery.go (hunk 1 above)
Comment (0xquark) [reply internal/pkg/networkscanner/portdiscovery/portdiscovery.go:31]: Can we work on the Cli on a separate PR?
Like i've made changes according to the go styling guide like package name should be the directory names. 
So i've change the path to `internal/pkg/networkscanner/` from `internal/pkg/network-scanner/`.

**Comment by 0xquark** [comment]
@slashben For CLI : https://github.com/spf13/cobra 
This might be helpful....I've seen it used in many big projects

**Comment by slashben** [comment]
@0xquark sure, my code is just an example!

The point is that I try to sort out the directory structure, give me 30m

**Comment by slashben** [comment]
Looks ok for me!

**Comment by 0xquark** [comment]
This PR is ready to be Merged 🚀

File: internal/pkg/networkscanner/portdiscovery/portdiscovery.go (hunk 1 above)
Comment (slashben) [reply internal/pkg/networkscanner/portdiscovery/portdiscovery.go:31]: Sure, just note this

**Review by slashben — APPROVED** [review]

**Pull request merged**